src/libraries/Microsoft.PowerFx.Interpreter/ReflectionFunction.cs

```diff
@@ -403,7 +405,20 @@ public async Task<FormulaValue> InvokeAsync(IServiceProvider serviceProvider, Fo
                 args2.Add(cancellationToken);
             }
 
-            var result = _info.Method.Invoke(this, args2.ToArray());
+            object result = default;
+            try
+            {
+                result = _info.Method.Invoke(this, args2.ToArray());
+            }
+            catch (TargetInvocationException e)
```

Comment:
>TargetInvocationException [](http://example.com/codeflow?start=19&length=25)

Can we double check with an async function?
(ie, there could be some AggregateException in there that needs to get unwrapped)

Reply:
You are correct, async would directly throw the `CustomFunctionErrorException` while unwrapping. Updated the code for the same and added the test.